src/libraries/System.Private.CoreLib/src/System/Text/Latin1Utility.cs

```diff
@@ -250,7 +233,7 @@ private static unsafe nuint GetIndexOfFirstNonLatin1Char_Sse2(char* pBuffer, nui
 
             // Now adjust the read pointer so that future reads are aligned.
 
-            pBuffer = (char*)(((nuint)pBuffer + SizeOfVector128InBytes) & ~(nuint)(SizeOfVector128InBytes - 1));
+            pBuffer = (char*)(((nuint)pBuffer + SizeOfVector128InBytes) & ~(SizeOfVector128InBytes - 1));
```

Comment:
Need to keep the `(nuint)` cast here. Otherwise the `a & ~b` results in `~b` being calculated before zero-extension, which could result in the high 32 bits of the address being dropped.

Reply:
Then Roslyn needs a fix here, as it states the cast is unnecessary.